src/fs/mkdir.rs

```diff
@@ -9,20 +9,20 @@ use tokio::fs;
 use crate::utils::result::ResultExt;
 
 pub async fn mkdir<'js>(ctx: Ctx<'js>, path: String, options: Opt<Object<'js>>) -> Result<String> {
-    let mut recursive = false;
-    let mut mode = 0o777;
-
-    if let Some(options) = options.0 {
-        recursive = options.get("recursive").unwrap_or_default();
-        mode = options.get("mode").unwrap_or(0o777);
-    }
+    let (recursive, mode) = get_mkdir_params(options);
 
     if recursive {
         fs::create_dir_all(&path).await
     } else {
         fs::create_dir(&path).await
     }
-    .or_throw_msg(&ctx, &format!("Can't create dir \"{}\"", &path))?;
+    .or_throw_msg(
+        &ctx,
+        &format!(
+            "EEXIST: file already exists, mkdir. Can't create dir \"{}\"",
```

Comment:
The error message returned by mkdirsync and mkdir in Rust is different and I wanted to align them with Node but you are right it could be another message. I fixed it.